thebeetoken/beenest-web
# Show driving directions

## Description
Business travelers care about being close to specific locations (conference halls, work sites, and so on) so let's help them out!

Note that this has been branched from #289 and should be reviewed/merged after that has been reviewed/merged.

## How to Test
1. Search (e.g. for "San Francisco")
2. Specify a "Close To..." filter (e.g. "Moscone Center")
3. Click on listings on the map
4. Expect to see the route to your destination appear

Which devices did you test on?

## REVIEWERS:
Check against these principles:

### High level
Does this code need to be written?
What are the alternatives?
Will this implementation become a support issue?
How much error margin does this solution have?

### Code
* Does the code follow industry standards?
JS: https://github.com/airbnb/javascript
React: https://github.com/airbnb/javascript/tree/master/react
https://github.com/vasanthk/react-bits
Documentation headers: http://usejsdoc.org/index.html

* Is there duplicated code? Can it be refactored into a shared method?
* Is the code consistent with our project?
* Are there unit tests? Do they test the states?
* Is the person refactoring another developer's code? If possible, did the original developer approve?

### Variables/Naming:
* Would the variable type led to future edge cases?
* Are the variable naming clear? Would the value contain something other than what the name describes.

### Security
* Can this be hacked or abused by the user?

## Further Work
Playing with this, there are a variety of UX improvements I might want to see. Think we should play around with this, discuss, and itemize any non-blockers with Jira tickets.

## Learnings
I'm just generally impressed with the API design of Google Maps. Easy to do hard things but not hard to do easy things!
